File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.util.*;
class Solution {
    public int[] solution(String today, String[] terms, String[] privacies) {
        int todayDate = getDate(today);
        int[] termsArray = new int[26];
        for(int i=0; i<terms.length; i++){
            int index = terms[i].charAt(0) - 'A';
            int term = Integer.parseInt(terms[i].split(" ")[1]) * 28;
            termsArray[index] = term;
        }
        List<Integer> list = new ArrayList<>();
        for(int i=0; i<privacies.length; i++){
            int startDate = getDate(privacies[i].split(" ")[0]);
            int categoryIndex = privacies[i].charAt(11) - 'A';
            int period = termsArray[categoryIndex];
            if(todayDate >= startDate + period){
                list.add(i+1);
            }
        }
        int[] answer = new int[list.size()];
        for(int i=0; i<list.size(); i++){
            answer[i] = list.get(i);
        }
        return answer;
    }
    
    public int getDate(String date){
        String[] arr = date.split("\\.");
        int year = Integer.parseInt(arr[0]);
        int month = Integer.parseInt(arr[1]);
        int day = Integer.parseInt(arr[2]);
        return year * 12 * 28 + month * 28 + day;
    }
}

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    def getDate(date):
        return int(date[0:4]) * 12 * 28 + int(date[5:7]) * 28 + int(date[8:10])
    todayDate = getDate(today)
    termsArray = [0] * 26
    for i in range(len(terms)):
        index = ord(terms[i][0:1]) - ord("A")
        term = int(terms[i][2:]) * 28
        termsArray[index] = term
    answer = []
    for i in range(len(privacies)):
        startDate = getDate(privacies[i][0:10])
        index = ord(privacies[i][11]) - ord('A')
        period = termsArray[index]
        if startDate + period <= todayDate:
            answer.append(i+1)
    return answer